src/documents/views/StudyFilesListView.js

```diff
@@ -42,6 +45,35 @@ const StudyListSkeleton = () => (
   </Grid>
 );
 
+const filterFiles = (fileList, filters) => {
+  const sortFuncs = {
+    createDate: createDateSort,
+    modifyDate: modifiedDateSort,
+    default: defaultSort,
+  };
+  var sortedList = fileList.sort(
+    sortFuncs[filters.sortMethod] || sortFuncs.default,
+  );
+  sortedList =
+    filters.sortDirection === 'ascending' ? sortedList : sortedList.reverse();
```

Comment:
I think this is why it takes two clicks to sort. It will only reverse when the value is `descending`.
Maybe move the direction out of the filter function and do it like this:
```
  const filteredFiles =                                                             
    filters.sortDirection === 'ascending'                                                                                                                                                                                                     
      ? filterFiles(files, filters)                                                 
      : filterFiles(files, filters).reverse();  
```